core/client/fs/src/test/java/alluxio/client/file/FileInStreamTest.java

```diff
@@ -31,33 +32,41 @@
 import org.junit.After;
 import org.junit.Assert;
 import org.junit.Before;
+import org.junit.Rule;
 import org.junit.Test;
 import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
 import org.mockito.Mockito;
 import org.mockito.invocation.InvocationOnMock;
 import org.mockito.stubbing.Answer;
 import org.powermock.api.mockito.PowerMockito;
 import org.powermock.core.classloader.annotations.PrepareForTest;
-import org.powermock.modules.junit4.PowerMockRunner;
+import org.powermock.modules.junit4.rule.PowerMockRule;
 
 import java.io.IOException;
 import java.nio.ByteBuffer;
 import java.util.ArrayList;
 import java.util.Arrays;
+import java.util.Collection;
 import java.util.List;
 
 /**
  * Tests for the {@link FileInStream} class.
+ *
+ * It is a parameterized test that checks different caching behaviors when the blocks are located at
+ * different locations.
  */
-@RunWith(PowerMockRunner.class)
+@RunWith(Parameterized.class)
```

Comment:
This is cool, I didn't know about `PowerMockRule` previously. I think using multiple rules with the `PowerMockRule` is worth a shot.